src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -780,5 +803,42 @@ private void CheckExplicitImplementationAccessor(MethodSymbol? thisAccessor, Met
                 diagnostics.Add(ErrorCode.ERR_ExplicitPropertyAddingAccessor, thisAccessor.GetFirstLocation(), thisAccessor, explicitlyImplementedEvent);
             }
         }
+
+        internal bool IsPartial => (this.Modifiers & DeclarationModifiers.Partial) != 0;
+
+        /// <summary>
+        /// <see langword="true"/> if this symbol corresponds to a semi-colon body declaration.
+        /// <see langword="false"/> if this symbol corresponds to a declaration with custom <see langword="add"/> and <see langword="remove"/> accessors.
+        /// </summary>
+        protected abstract bool IsFieldLike { get; }
```

Comment:
>IsFieldLike [](http://example.com/codeflow?start=32&length=11)

This name feels confusing to me since "field-like" has a specific meaning in the spec as an event that can be used like a field within the containing type, and which is not the case for partial events.

If this property just intended for use in `IsPartialDefinition` and `IsPartialImplementation`, would it make sense to rename this to something more specific? For instance, match `SourcePropertySymbol` and name this `AccessorsHaveImplementation`.